### ISSUE TITLE
Research: How to automate a list of all existing HAL Links

### ISSUE THREAD

**Issue by roamye** [project-lux/lux-middletier#59]
**Problem Description:**
When the HAL LINK test cases were created it only captured the current existing HAL LINKS (last updated in March 2024).
Since then we have created/removed many HAL LINKS and need to capture these new ones in our regression testing . Our current process consists of (1)  manually generate a list of existing HAL links and (2) manually generate the "search will match' URL for each corresponding HAL LINK. 

As more HAL Links will be created/removed it would be easier to automate these two processes to allow BA Amy to add/remove HAL links to the regression testing. 

This ticket serves as research on how we can automate this process.

**Expected Behavior/Solution:**
We wish to automate the generation of list which includes existing HAL links and its corresponding 'search will match' link. 

**Requirements:**
TBD

**Needed for promotion:**
If an item on the list is not needed, it should be crossed off but not removed.

- [ ] Wireframe/Mockup - Mike
- [ ] Committee discussions - Sarah
- [ ] Feasibility/Team discussion - Sarah
- [ ] Backend requirements - TBD
- [ ] Frontend requirements- TBD
- [ ] Questions
- List of questions for discussions. Answers should be documented within the issue.

**UAT/LUX Examples:**

- Example endpoints from LUX to use for development and testing, if applicable.

**Dependencies/Blocks:**

- Blocked By: Issues that are blocking the completion of the current issue.
- Blocking: Issues being blocked by the completion of the current issue.

**Related Github Issues:**

- Issues that contain similar work but are not blocking or being blocked by the current issue.

**Related links:**

- These links can consist of resources, bugherds, etc.

**Comment by zachjschwartz** [comment]
Per 9/18 TF: does not seem to be captured by regression testing updates. Ticket still needed. Can be automated but requires manual updating after a new dataset.

**Comment by zachjschwartz** [comment]
On hold pending backend discussions (9/25 UAT)
